### PR TITLE
docs(core): in template syntax guide, make SVG example more clear

### PR DESCRIPTION
**Pull request by Splaktar**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
The current naming may be confusing to some users: `[attr.fill]="fill"`.

Issue Number: Relates to #30559

## What is the new behavior?
The naming is less potentially confusing: `[attr.fill]="fillColor"`

- add e2e test for SVG template example
- add SVG template component to example app
- fix template syntax example app
  - linting errors
  - runtime exceptions
  - template type errors
  - deprecated type casting
  - deprecated currency pipe example

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
PR suggested by @gkalpak in https://github.com/angular/angular/pull/30559#discussion_r298796259.

### CODE REVIEW

**Comment by kapunahelewong** [comment]
Thank you for this, @Splaktar!

File: aio/content/examples/template-syntax/src/app/svg.component.svg [hunk 1]
```diff
@@ -1,6 +1,10 @@
-<svg>
-  <g>
-    <rect x="0" y="0" width="100" height="100" [attr.fill]="fill" (click)="changeColor()" />
-    <text x="120" y="50">click the rectangle to change the fill color</text>
-  </g>
-</svg>
+<div id="svg-template">
```
Comment (gkalpak) [inline aio/content/examples/template-syntax/src/app/svg.component.svg]: Why this change? (It is confusing imo, since the example was trying to demonstrate (among other things), that you can use pure SVG files as templates.)
Also, with this change we would be showing one thing in the guide (the SVG-only docregion), but test against another (the whole file) 😞

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @Splaktar! Great clean-up on the side ❤️

I left some minor comments and questions.

Can you, also, change the commit message to use `docs-infra` or `core` as the scope (we have moved away from the `aio` commit message scope, because it was a little cryptic). If you want to indicate that this is a docs change to the template syntax guide (and the associated examples), the use `docs(core): ...`. If you want to indicate that this is a fix to the docs infrastructure (e.g. fixing errors in the example), use `fix(docs-infra): ...`. (It is kind of hard to decide atm, because there is one commt that does it all, so maybe you could split it up  into more commits :wink:)

File: aio/content/examples/template-syntax/src/app/app.component.html [hunk 2]
```diff
@@ -442,7 +443,7 @@ <h3>Result: {{currentHero.name}}</h3>
 
 <!-- #docregion without-NgModel -->
 <input [value]="currentHero.name"
-       (input)="currentHero.name=$event.target.value" >
+       (input)="updateCurrentHeroName($event)">
```
[the remark this answers is not in the document]
Comment (Splaktar) [reply aio/content/examples/template-syntax/src/app/app.component.html]: Template type checking error (`Unresolved variable value`) due to `value` in `$event.target.value` not being a defined type on `EventTarget`.

File: aio/content/examples/template-syntax/src/app/svg.component.svg (hunk 1 above)
Comment (Splaktar) [reply aio/content/examples/template-syntax/src/app/svg.component.svg]: Good points.

I was trying to keep the `id="svg-template"` off of the `<svg>` element and thus not visible on the docs page, but also allow the e2e test to work without having to query for something fragile like `by.css('svg')`.

I guess that I can put this `div` and `id` in the template of the app (`app.component.html`) instead.

Comment (Splaktar) [reply aio/content/examples/template-syntax/src/app/svg.component.svg]: Removed `id="svg-template"`. Just querying for `by.css('app-svg')` now.

**Review by gkalpak — APPROVED** [review]
Awesome! Thx, @Splaktar :tada:

**Comment by gkalpak** [comment]
**merge-assistance**: Global approval for this docs-only change.
